aim/db/hashtree_db_listener.py

```diff
@@ -36,44 +36,64 @@ def on_commit(self, session, added, updated, deleted):
         # Segregate updates by tenant
         updates_by_tenant = {}
         all_updates = [added, updated, deleted]
+        conf = tree_model.CONFIG_TREE
+        monitor = tree_model.MONITORED_TREE
+        oper = tree_model.OPERATIONAL_TREE
         for idx in range(len(all_updates)):
+            tree_index = 0 if idx < 2 else 1
             for res in all_updates[idx]:
                 key = self.tt_maker.get_tenant_key(res)
                 if not key:
                     continue
-                updates_by_tenant.setdefault(key, ([], [], [], []))
-                updates_by_tenant[key][
-                    (0 if idx < 2 else 1) +
-                    (2 if isinstance(res, aim_status.OperationalResource)
-                     else 0)].append(res)
+                updates_by_tenant.setdefault(
+                    key, {conf: ([], []), monitor: ([], []), oper: ([], [])})
+                if isinstance(res, aim_status.OperationalResource):
+                    # Operational Tree
+                    updates_by_tenant[key][oper][tree_index].append(res)
+                elif getattr(res, 'monitored', None):
+                    # Monitored Tree
+                    updates_by_tenant[key][monitor][tree_index].append(res)
+                else:
+                    # Configuration Tree
+                    updates_by_tenant[key][conf][tree_index].append(res)
 
         # Query hash-tree for each tenant and modify the tree based on DB
         # updates
         class DummyContext(object):
             db_session = session
         ctx = DummyContext()
 
-        upd_trees, udp_op_trees = [], []
+        upd_trees, udp_op_trees, udp_mon_trees = [], [], []
         for tenant, upd in updates_by_tenant.iteritems():
             try:
-                ttree = self.tt_mgr.get(ctx, tenant)
-                ttree_operational = self.tt_mgr.get(ctx, tenant,
-                                                    operational=True)
+                ttree = self.tt_mgr.get(ctx, tenant, tree=conf)
+                ttree_operational = self.tt_mgr.get(ctx, tenant, tree=oper)
+                ttree_monitor = self.tt_mgr.get(ctx, tenant, tree=monitor)
             except hexc.HashTreeNotFound:
                 ttree = htree.StructuredHashTree()
                 ttree_operational = htree.StructuredHashTree()
```

Comment:
It's not enforced by the DB per se, but if you do that through the TreeManager then it will be the case:

https://github.com/noironetworks/aci-integration-module/blob/monitored-universe/aim/db/tree_model.py#L106-L124